sdstore/cache2disk.go

```diff
@@ -1,21 +1,230 @@
 package sdstore
 
 import (
+	"encoding/json"
 	"fmt"
 	"github.com/otiai10/copy"
+	"io/ioutil"
+	"log"
 	"os"
 	"path/filepath"
+	"reflect"
 	"strings"
 )
 
+// taken and modified from https://stackoverflow.com/questions/32482673/how-to-get-directory-total-size
+/*
+get directory size recursive in bytes
+param - path         		directory path
+return - size in bytes
+*/
+func getDirSizeInBytes(path string) int64 {
+	sizes := make(chan int64)
+	readSize := func(path string, file os.FileInfo, err error) error {
+		if err != nil || file == nil {
+			return nil
+		}
+		if !file.IsDir() {
+			sizes <- file.Size()
+		}
+		return nil
+	}
+
+	go func() {
+		_ = filepath.Walk(path, readSize)
+		close(sizes)
+	}()
+
+	size := int64(0)
+	for s := range sizes {
+		size += s
+	}
+
+	return size
+}
+
+/*
+compare md5 for source and destination directories
+param - src         		source directory
+param - dest			destination directory
+return - bytearray / error   	return md5 byte array, error - md5-same / md-chagned
+*/
+func checkMd5(src, dest string) ([]byte, error) {
+	var oldMd5 map[string]string
+	var newMd5 map[string]string
+
+	fmt.Println("start md5 check")
```

Comment:
we should standardize on logging in go modules. eg: https://github.com/sirupsen/logrus so that we can control log levels and modules etc

Reply:
we could address this later.

Reply:
yes will address it later

Reply:
did initial implementation of logging module for cache2disk